mock_ros2_project/publisher_node.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+"""
+ROS2 Publisher Node Example
+
+This node demonstrates how to create a simple publisher that sends
+string messages at regular intervals to demonstrate basic pub/sub communication.
+"""
+
+import rclpy
+from rclpy.node import Node
+from std_msgs.msg import String
+
+
+class PublisherNode(Node):
+    """Simple publisher node that publishes string messages.
+    
+    This node creates a publisher that sends string messages containing
+    a counter value at regular intervals.
+    """
+    
+    def __init__(self) -> None:
+        """Initialize the publisher node."""
+        super().__init__('publisher_node')
+        
+        # Create publisher for string messages
+        self.publisher_ = self.create_publisher(String, 'send_topic', 10)
```

Comment:
Consider exposing the topic name and QoS depth as configurable ROS parameters rather than hard-coding them. This will make the node reusable in different deployments without source modification and allow launch files to set appropriate values.

```suggestion
        topic_name = self.declare_parameter('topic_name', 'send_topic').get_parameter_value().string_value
        qos_depth = self.declare_parameter('qos_depth', 10).get_parameter_value().integer_value
        self.publisher_ = self.create_publisher(String, topic_name, qos_depth)
```